src/click_outside.rs

```diff
@@ -0,0 +1,398 @@
+    {
+        // Check if element has been removed from DOM
+        if element.parent_node().is_none() {
+            return;
+        }
+
+        // Check for specific class names to ignore
+        let class_name = element.class_name();
+        if class_name == "tag-select-edit"
+            || class_name == "tag-select-edit-done"
+            || class_name == "tag-input-close"
+        {
+            return;
+        }
+
+        // Check if click is inside any of the select elements
+        let select_classes = ["searchable-select", "mini-select", "tag-group-input-outer"];
+        for class in &select_classes {
+            let elements = document.get_elements_by_class_name(class);
+            for i in 0..elements.length() {
+                if let Some(select_elem) = elements.item(i)
+                    && is_descendant_of(element, &select_elem)
+                {
+                    return;
+                }
+            }
+        }
+
+        // Close all dropdowns
+        let dropdown_classes = [
+            "searchable-select-list-down",
+            "mini-select-list-down",
+            "tag-group-input-select",
+        ];
+        for class in &dropdown_classes {
+            let elements = document.get_elements_by_class_name(class);
+            for i in 0..elements.length() {
+                if let Some(elem) = elements.item(i)
+                    && let Ok(html_elem) = elem.dyn_into::<HtmlElement>()
+                {
+                    let _ = html_elem.style().set_property("display", "none");
+                }
+            }
+        }
+
+        remove_listen_click();
+    }
+}
+
+fn close_custom_select_complex(event: &MouseEvent, document: &Document) {
+    // Check if the target's parent node exists
+    if let Some(target) = event.target()
+        && let Some(element) = target.dyn_ref::<Element>()
+    {
+        // Check if element has been removed from DOM or doesn't exist in body
+        if element.parent_node().is_none() {
+            return;
+        }
+
+        // Check if click is inside any complex select
+        let elements = document.get_elements_by_class_name("complex-select");
+        for i in 0..elements.length() {
+            if let Some(select_elem) = elements.item(i)
+                && is_descendant_of(element, &select_elem)
+            {
+                return;
+            }
+        }
+
+        // Close all complex select popups
+        let elements = document.get_elements_by_class_name("complex-select-pop");
+        for i in 0..elements.length() {
+            if let Some(elem) = elements.item(i)
+                && let Ok(html_elem) = elem.dyn_into::<HtmlElement>()
+            {
+                let _ = html_elem.style().set_property("display", "none");
+            }
+        }
+
+        remove_listen_click_complex();
+        remove_listen_mousedown_complex();
+    }
+}
+
+fn is_descendant_of(element: &Element, ancestor: &Element) -> bool {
```

Comment:
Please check whether the behavior of `is_descendant_of` can be replaced with the built-in function `Node.contains()`.